### DISCUSSION TITLE
How to plot 2 ggplots side by side?

### DISCUSSION THREAD

**Discussion by WillemSleegers** [quarto-dev/quarto-cli#590] [Q&A] [answered]
Hi everyone,

Does anyone know how to present two ggplots side by side using the 2 column layout and also have only 1 single code chunk displayed before it?

If I simply plot two ggplots, I can get them side by side, but I then see 2 code chunks right above the plots, one for each plot.

**Comment by BradyAJohnston** [comment, marked as answer]
Testing it out, I get a single code chunk above the two plots that are side by side: 
````r
# Testing References

```{r}
#| label: fig-TESTING
#| fig-cap: "A test plot."
#| layout: "[1, 1]"
library(ggplot2)
plt <- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) + 
  geom_point(size = 3) + 
  theme_classic()

plt
plt2 <- ggplot(airquality, aes(Wind, Temp)) + 
  geom_point() + 
  theme_classic()

plt2
```
````

The code above results in this when rendered: 
![image](https://user-images.githubusercontent.com/36021261/162558809-0856e056-1d95-499d-a2ec-36e8875a6190.png)

**Reply by jjallaire** [reply]
Note that `layout` here can be simplified to `layout-nol=2`

**Comment by WillemSleegers** [comment]
The code works, thanks!

Just to clarify, you need a label and the label should start with `fig`?

**Reply by BradyAJohnston** [reply]
If the output is a figure, yes! If the output is a table it should start with `tbl`. This tripped me up when I was first trying to figure things out.

**Reply by WillemSleegers** [reply]
That's good to know!

**Comment by jjallaire** [comment]
Saw this discussion and I think its unfortunate that you need to add crossrefs in order to get this to work. Just made a change to our layout logic that will automatically fuse the code blocks that feed the layout: https://github.com/quarto-dev/quarto-cli/commit/274b86cc360801087ffbae2337e048fea768039f

So this code will work as of v0.9.251 (building now)

````markdown
```{r}
#| layout: "[1, 1]"
library(ggplot2)
plt <- ggplot(mtcars, aes(mpg, wt, colour = factor(cyl))) + 
  geom_point(size = 3) + 
  theme_classic()

plt
plt2 <- ggplot(airquality, aes(Wind, Temp)) + 
  geom_point() + 
  theme_classic()

plt2
```
````

**Reply by BradyAJohnston** [reply]
Excellent! Thanks for continuous and quick improvements!
